functions/Export-DbaScript.ps1

```diff
@@ -154,19 +154,19 @@ function Export-DbaScript {
         $commandName = $MyInvocation.MyCommand.Name
         $prefixArray = @()
 
-        $appendToScript = $Append
+        # if the Append switch is used then ensure the scripting options are updated
+        $appendToScript = $false
+        if ($Append.IsPresent) {
+            $ScriptingOptionsObject.AppendToFile = $true
+            $appendToScript = $true
+        }
+
         if ($ScriptingOptionsObject) {
             # Check if BatchTerminator is consistent
             if (($($ScriptingOptionsObject.ScriptBatchTerminator)) -and ([string]::IsNullOrWhitespace($BatchSeparator))) {
                 Write-Message -Level Warning -Message "Setting ScriptBatchTerminator to true and also having BatchSeperarator as an empty or null string may produce unintended results."
             }
-
-            if ($ScriptingOptionsObject.AppendToFile) {
-                Write-Message -Level Verbose -Message "The ScriptingOptionsObject AppendToFile setting of true will override Append parameter."
-                $appendToScript = $true
-            }
```

Comment:
Why are we removing the verbose message here?

Reply:
My thought here was that the Append param should be the overriding factor instead of the ScriptingOptionsObject. From the caller's perspective I think this is more intuitive. Using the -Append param wouldn't be overwritten by another object's state.